### PR TITLE
Extend endpoint related interfaces

### PR DESCRIPTION
**Pull request by aditighag**
See commit description.

### CODE REVIEW

File: pkg/hubble/parser/getters/getters.go
Comment (christarazi) [inline pkg/hubble/parser/getters/getters.go]: Maybe worth documenting that this interface gets you `policy.Endpoint` rather than the "real" endpoints `endpoint.Endpoint`?

**Review by christarazi — APPROVED** [review]
LGTM, minor optional comment if it makes sense

File: pkg/hubble/parser/getters/getters.go
Comment (aditighag) [reply pkg/hubble/parser/getters/getters.go]: Yup, forgot to add documentation.

File: pkg/policy/resolve_test.go
Comment (twpayne) [inline pkg/policy/resolve_test.go:54]: I think this only declares that `DummyEndpoint` implements these methods, it does not actually provide any implementations of them. This is what's causing the panic in the Travis build:

```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1a17733]
goroutine 1117 [running]:
github.com/cilium/cilium/pkg/policy.(*dummyEndpoint).IsHost(0xc000ff6d80, 0x2e3dd60)
	<autogenerated>:1 +0x33
github.com/cilium/cilium/pkg/policy.ruleSlice.updateEndpointsCaches(0xc000fc0000, 0x3e8, 0x3e8, 0x21061e0, 0xc000ff6d80, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/rules.go:165 +0x123
github.com/cilium/cilium/pkg/policy.ruleSlice.UpdateRulesEndpointsCaches.func1(0x21061e0, 0xc000ff6d80)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/repository.go:446 +0x85
github.com/cilium/cilium/pkg/policy.(*EndpointSet).ForEachGo.func1(0xc0007640f0, 0xc000a1ead0, 0x21061e0, 0xc000ff6d80)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/identifier.go:71 +0x42
created by github.com/cilium/cilium/pkg/policy.(*EndpointSet).ForEachGo
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/policy/identifier.go:70 +0x13f
```

In this case, there is no actual implementation of `IsHost` for `dummyEndpoint`.

File: pkg/policy/identifier_test.go
Comment (twpayne) [inline pkg/policy/identifier_test.go:29]: This only declares that `DummyEndpoint` implements all the methods of `Endpoint`, but does not add any implementation for them.

File: pkg/policy/resolve_test.go
Comment (twpayne) [reply pkg/policy/resolve_test.go:54]: Alternatively, when the `dummyEndpoint` is created, its `Endpoint` value needs to be set to something that includes an  implementation of `IsHost`.

Comment (aditighag) [reply pkg/policy/resolve_test.go:54]: Thanks for the review. Yes, the main motivation here is that users are not forced to implement all the method of an interface, rather only the ones that are relevant to the unit test cases. 
I'll triage the Travis failure (looks like the test passed in my local environment). Thanks!

**Comment by aditighag** [comment]
Travis CI run failed because of the rate limit issue. 

>anyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit

File: pkg/policy/resolve_test.go
Comment (aditighag) [reply pkg/policy/resolve_test.go:54]: As it turns out, `IsHost()` method was needed in one of the test permutations that Travis does. 

>Endpoint         // Implement methods of the interface that need to mock out real behavior.

Based on the comment, implementing the method.

**Comment by aditighag** [comment]
test-me-please

**Review by twpayne — APPROVED** [review]

**Comment by aditighag** [comment]
GKE test failed due to failure to scale - https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/4130/

**Comment by aditighag** [comment]
test-gke

**Comment by aditighag** [comment]
test-gke

**Review by twpayne — APPROVED** [review]

**Comment by aanm** [comment]
Blocking merging for now to discuss some fundamentals around this PR.

**Review by aanm — APPROVED** [review]

**Review by tklauser — APPROVED** [review]

**Pull request merged**
